core/state/state_object.go

```diff
@@ -305,8 +310,10 @@ func (s *stateObject) updateTrie(db Database) Trie {
 
 // UpdateRoot sets the trie root to the current root hash of
 func (s *stateObject) updateRoot(db Database) {
-	s.updateTrie(db)
-
+	if s.updateTrie(db) == nil {
+		// No changes, storage trie is not even loaded
```

Comment:
I prefer to put these kinds of comments outside the if check, makes things more readable, as I know what the following code is, vs. entering the if and then figuring it out why I did it in the first place:

```
// If nothing changed, don't bother with hashing anything
```